temporalio/worker/workflow_instance.py

```diff
@@ -1038,6 +1050,24 @@ def _next_seq(self, type: str) -> int:
         self._curr_seqs[type] = seq
         return seq
 
+    def _register_task(self, task: asyncio.Task, *, name: Optional[str]) -> None:
+        # Name not supported on older Python versions
+        if sys.version_info >= (3, 8):
+            # Put the workflow info at the end of the task name
+            name = name or task.get_name()
+            name += f" (workflow: {self._info.workflow_type}, id: {self._info.workflow_id}, run: {self._info.run_id})"
+            task.set_name(name)
+        # Add to and remove from our own non-weak set instead of relying on
+        # Python's weak set which can collect these too early
+        self._tasks.add(task)
+        task.add_done_callback(self._tasks.remove)
+        # When the workflow is GC'd (for whatever reason, e.g. eviction or
+        # worker shutdown), still-open tasks get a message logged to the
+        # exception handler about still pending. We disable this if the
+        # attribute is available.
+        if hasattr(task, "_log_destroy_pending"):
+            setattr(task, "_log_destroy_pending", False)
```

Comment:
Should we only do this for workflows that we know we've evicted or after shutdown?

Reply:
I think this only applies in those two cases since this is only used on GC and I hold a strong reference until eviction/shutdown or the task is done. I just choose to do this at registration time instead of loop over all tasks at the end.